python/cudf/cudf/core/scalar.py

```diff
@@ -321,3 +376,7 @@ def _dispatch_scalar_unaop(self, op):
 
     def astype(self, dtype):
         return Scalar(self.value, dtype)
+
+    @classmethod
+    def _clear_cache(cls):
+        type(cls).clear(cls)
```

Comment:
It seems like we might mean:
```suggestion
        cls.clear()
```

Both appear to be equivalent. https://godbolt.org/z/7GzzKnhjf

Reply:
Hmm, they shouldn't always resolve to exactly the same thing though. For example: https://godbolt.org/z/MG96rP56c

Reply:
I believe the better practice here would be to invoke the method from `type(cls)`, rather than `cls`, for kind of the same reason you don't want to invoke classmethods from instances of regular classes (even though you technically _can_).

Reply:
I think your intuition that something here is smelly is correct.

On the one hand I wanted `Scalar` to be totally oblivious to instance caching, but on the other hand, I added a `_clear_cache()` classmethod to it. That now feels like the wrong approach, and is probably why this code looked so awkward.

Instead, I decided to go with something closer to your initial suggestion. I changed the name of the metaclass method `clear()` to  `_clear_instance_cache()`,  with the intent to use it directly from subclasses and not have to worry about naming conflicts.

TL;DR, to clear `Scalar` instances, one can now just do `Scalar._clear_instance_cache()`.

Reply:
I'm quite happy with this outcome. Nice solution, @shwina!